File: Kekom-RSA/rsa.js
function RSA() {
  var gcd, p, q, no, n, t, e, i, x, d, ct, ctt, dtt, dt;
  
  // Helper function to calculate GCD
  gcd = function (a, b) { 
      return (!b) ? a : gcd(b, a % b); 
  };
  
  // Get input values
  p = parseInt(document.getElementById('p').value);
  q = parseInt(document.getElementById('q').value);
  no = parseInt(document.getElementById('msg').value);
  
  // Calculate n and t
  n = p * q;
  t = (p - 1) * (q - 1);
  
  // Find a valid exponent e
  for (e = 2; e < t; e++) {
      if (gcd(e, t) == 1) {
          break;
      }
  }

  // Find the private key d
  for (i = 0; i < 10; i++) {
      x = 1 + i * t;
      if (x % e == 0) {
          d = x / e;
          break;
      }
  }

  // Encryption (c = m^e mod n)
  ctt = Math.pow(no, e).toFixed(0);
  ct = ctt % n;

  // Decryption (m = c^d mod n)
  dtt = Math.pow(ct, d).toFixed(0);
  dt = dtt % n;

  // Display results
  document.getElementById('publickey(N)').innerHTML = n;
  document.getElementById('exponent(e)').innerHTML = e;
  document.getElementById('privatekey(d)').innerHTML = d;
  document.getElementById('ciphertext(ct)').innerHTML = ct;
}

File: Kekom-RSA/rsa.c
#include<stdio.h> 
#include<math.h> 

int gcd(int a, int h) 
{ 
    int temp; 
    while (1) 
    { 
        temp = a % h; 
        if (temp == 0) 
            return h; 
        a = h; 
        h = temp; 
    } 
} 

int main() 
{ 
    float p = 3; 
    float q = 7; 
    
    float n = p * q; 
    
    float e = 2; 
    float phi = (p - 1) * (q - 1); 
    while (e < phi) 
    { 
        if (gcd(e, phi) == 1) 
            break; 
        else
            e++; 
    } 
    
    int k = 2;
    float d = (1 + (k * phi)) / e; 
    
    float msg = 20; 
    
    printf("Message data = %f", msg); 
    
    float c = pow(msg, e); 
    c = fmod(c, n); 
    printf("\nEncrypted data = %f", c); 
    
    float m = pow(c, d); 
    m = fmod(m, n); 
    printf("\nOriginal Message Sent = %f", m); 
    
    return 0; 
}
